sherpa/astro/flux.py

```diff
@@ -529,12 +529,91 @@ def sample_flux(fit, data, src,
 
 def calc_sample_flux(id, lo, hi, session, fit, data, samples, modelcomponent,
                      confidence):
+    """Given a set of parameter samples, estimate the flux distribution.
+
+    This is similar to sample_flux but returns values for both the full
+    and component models (which can be the same). The set of parameter
+    samples is searched to remove rows where the parameters lie outside
+    the soft limits, and then this set of rows is used to calculate the
+    flux of the modelcomponent expression (the flux for the full model
+    is taken from the samples argument). The resulting set of fluxes is
+    used to calculate the median and the confidence range.
+
+    .. versionchanged:: 4.13.1
+       The clipping now includes parameters at the soft limits;
+       previously they were excluded which could cause problems with
+       parameters for which we can only calculate an upper limit.
+
+    Parameters
+    ----------
+    id : int or str
+        The dataset identifier. It must exist and have an associated model in
+        session.
+    lo : number or None, optional
+        The lower edge of the dataspace range for the flux calculation.
```

Comment:
I have not come across the term "dataspace" a lot as a sherpa user. I know we use it internally in specs, in the code, and in our discussions, but I'm not sure this wording is helpful for the user. I don't have any better ideas though... I mean, could this number even be outside the range of the datagrid? Maybe a functional explanation "Use this parameter to do X" would be helpful in addition to the current text.

Reply:
Or is that something for later? (Since @DougBurke pulled this out of a larger change).

Reply:
This text was taken from the existing documentation from other functions in sherpa.astro.flux, so it is at least consistent. I bet if you look through other bits of code you would find this elsewhere.

I am reluctant to spend significant amount of time on tweaking and discussing documentation in these low-level routines without an effort to improve the "intended" user-layer code (e.g. the ui modules). Any change we make for these would need to be reflected in all those routines.

https://github.com/sherpa/sherpa/blob/b7cb202c27951c6dc2ee2ef0a2b661f1ff80c0db/sherpa/astro/ui/utils.py#L13040-L13042 which is the sherpa.astro.ui.sample_energy_flux routine uses the term "data grid" rather than "dataspace", bt I would argue that `dataspace` is a Sherpa term, since we have `sherpa.astro.ui.dataspace1d` - e.g. https://github.com/sherpa/sherpa/blob/b7cb202c27951c6dc2ee2ef0a2b661f1ff80c0db/sherpa/astro/ui/utils.py#L567

Reply:
You asked for a review of this PR, not all docstring in the entire sherpa package, so I give you one ;-) 
But that makes sense, fine with me.